replications/internal/queue_management.go

```diff
@@ -199,10 +190,8 @@ func (rq *replicationQueue) SendWrite(dp func([]byte) error) (int, error) {
 
 	// Advance the queue pointer forward, dropping data that has already been processed.
 	// TODO Are there errors from calling Advance() that can be more gracefully handled?
-	if _, err := scan.Advance(); err != nil {
-		return count, err
-	}
-	return count, nil
+	_, err = scan.Advance()
+	return err != nil && err != io.EOF
```

Comment:
This should log the `err` if it's not nil and not `io.EOF`, and return `false`. If err == nil then return `true`